main.go

```diff
@@ -65,6 +69,7 @@ func init() {
 
 	_ = solrv1beta1.AddToScheme(scheme)
 	_ = zkv1beta1.AddToScheme(scheme)
+	_ = certv1.AddToScheme(scheme)
 
 	// +kubebuilder:scaffold:scheme
 	flag.BoolVar(&useZookeeperCRD, "zk-operator", true, "The operator will not use the zk operator & crd when this flag is set to false.")
```

Comment:
As far as I've seen, calling `AddToScheme` works even if the Cert-manager CRDs are not installed so not sure a flag is technically needed but doesn't hurt to add one.

Reply:
Add to scheme is fine, the boolean is used when setting up the controller and determining whether to set "Owns" for that resource.

Reply:
Ok I see, I was confused because you put the comment on the `AddToScheme` line so thought the problem was about that line of code.

I don't think the Solr operator needs to own `Certificate` objects ... all it cares about is the TLS secret that gets created by the cert-manager in response to a change to the `Certificate`. It seems like the secret changing does trigger a reconcile in my testing but maybe we need to add a specific watch for that secret changing like you did for user-provided ConfigMaps?

Reply:
Ahh sorry for the confusion.

Yeah, it looks like Solr Operator is creating its own Secrets as well as finding secrets created by CertManager. If that's the case, then I think we will need both "Owns" and "Watches" with similar logic to the ConfigMaps. But I may be wrong there, this is uncharted territory.

Are we sure we don't need to own Certificates? We wait for them to come online, so we want to be notified when they have condition changes, right?

Reply:
From a reconcile perspective, we really only care about the TLS secret that cert-manager creates once the Certificate is issued. The "watching" of the Certificate to come online is really for status reporting while the cert is issuing as it can take several minutes for the cert to be issued. Notice the `isCertificateReady` is mostly about checking for the TLS secret.

The operator does create a Certificate for `autoCreate` mode but in that case, the cert definition should come from the SolrCloud CRD and we don't want to let users edit the Certificate externally; this is similar to the default `solr.xml` ConfigMap and any direct edits to that cm are lost, same with `autoCreate` certs.